docs/dev_guide/contents/extending_fido.rst

```diff
@@ -0,0 +1,501 @@
+
+* **Converters**: Adding a converter to the walker adds the function to both the creator and the applier.
+  For the VSO client this is used to convert each supported attr into a `~sunpy.net.attr.ValueAttr` which is then later processed by the appliers and creators.
+  This pattern can be useful if you would otherwise have to repeat a lot of logic in each of the applier functions for each type of Attr you support.
+
+An Example of ``AttrWalker``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this example we will write a parser for some simple queries which uses `~sunpy.net.attr.AttrWalker` to convert the query to a `dict` of URL query parameters for a HTTP GET request.
+Let's imagine we have a web service which you can do a HTTP GET request to ``https://sfsi.sunpy.org/search`` for some imaginary data from an instrument called SFSI (SunPy Fake Solar Instrument).
+This GET request takes three query parameters ``startTime``, ``endTime`` and ``level``, so a request might look something like: ``https://sfsi.sunpy.org/search?startTime=2020-01-02T00:00:00&endTime=2020-01-02T00:00:00&level=1``.
+Which would search for level one data between 2020-01-01 and 2020-01-02.
+
+As `~sunpy.net.attrs` has `~sunpy.net.attrs.Time` and `~sunpy.net.attrs.Level` we don't need to define any of our own attrs for this client.
+We do however want to write our own walker to convert them to the form out client's ``search()`` method wants to send them to the server.
+
+The first step is to setup the walker and define a creator method which will return a list of dicts, one for each independent search.
+
+.. code-block:: python
+
+    import sunpy.net.atrrs as a
+    from sunpy.net.attr import AttrWalker, AttrAnd, AttrOr, DataAttr
+
+    walker = AttrWalker()
+
+    @walker.add_creator(AttrOr)
+    def create_or(wlk, tree):
+        results = []
+        for sub in tree.attrs:
+            results.append(wlk.create(sub))
+
+        return results
+
+    @walker.add_creator(AttrAnd, DataAttr)
+    def create_and(wlk, tree):
+        result = dict()
+        wlk.apply(tree, result)
+        return [result]
+
+
+The call ``wlk.apply(...)`` inside the creator will walk any nested attrs and add their values to the dictionary as defined by the applier registered to each attr type.
+If we want our client to support searching by ``a.Time`` and ``a.Level`` as in the URL example above, we would need to register an applier for each of these attrs.
+
+.. code-block:: python
+
+    @walker.add_applier(a.Time)
+    def _(wlk, attr, params):
+        return params.update({'startTime': attr.start.isot,
+                              'endTime': attr.end.isot})
+
+    @walker.add_applier(a.Level)
+    def _(wlk, attr, params):
+        return params.update({'level': attr.value})
+
+
+This combination of creators and appliers would allow support any combination of queries consisting of ``a.Time`` and ``a.Level``.
```

Comment:
```suggestion
This combination of creators and appliers would allow support of any combination of queries consisting of ``a.Time`` and ``a.Level``.
```